### ISSUE TITLE
Calling clipper.deleteAll() breaks plans view (outlines/edges)

### ISSUE THREAD

**Issue by maidi29** [ThatOpen/engine_components#413]
### Describe the bug 📝

When calling `clipper.deleteAll()` any time before entering the plans view (like in the plans component tutorial), the outlines of the elements are broken. Something in `deleteAll` destroys the edges/outlines. What do I need to call to update them again?
The plans are still there, just the color of the meshes, and the edges thickness is not considered.

### Steps to reproduce 🔢
1. follow the tutorial for plans component
2. follow the tutorial for clipper
3. call `clipper.deleteAll()`
4. enter the plans view

_No response_

### System Info 💻

```shell
System:
    OS: macOS 14.5
    CPU: (11) arm64 Apple M3 Pro
    Memory: 100.80 MB / 18.00 GB
  Binaries:
    Node: 20.10.0 - ~/.nvm/versions/node/v20.10.0/bin/node
    npm: 10.2.3 - ~/.nvm/versions/node/v20.10.0/bin/npm
  Browsers:
    Chrome: 126.0.6478.56
```

### Used Package Manager 📦

npm

### Validations ✅

- [X] Read the [docs](https://docs.thatopen.com/intro).
- [X] Check that there isn't [already an issue](https://github.com/ThatOpen/engine_components/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a repository issue and not a framework-specific issue. For example, if it's a THREE.js related bug, it should likely be reported to [mrdoob/threejs](https://github.com/mrdoob/three.js) instead.
- [X] Check that this is a concrete bug. For Q&A join our [Community](https://people.thatopen.com/).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by agviegas** [comment]
Hey @maidi29 right now the plans use the clipper for cutting the model. We could skip the floorplan planes, but that feels a bit rigid. What if instead we allow to define a `type` property for planes? By default, the type would be `default`, but for plans it would be `plan`. Then, we could add a filter optional argument to `deleteAll()`, so it would look like this:

```ts
clipper.deleteAll({type: "default"});
```

Would this solve your issue? I'm trying to find a way that also support future use cases for clipping planes (sections, specific views of model details, etc).

**Comment by maidi29** [comment]
@agviegas thanks for the fast reply! Yes, I think this would solve my issue :) I only want to remove user-made clipping planes before entering the plans view but without having the styles of edges and meshes "destroyed"

**Comment by agviegas** [comment]
Solved from version `@thatopen/components@2.1.3` and `@thatopen/components-front@2.1.4`. Now all planes have a `type` string property that can be customized. The default value is `default`, and the planes created by the plan component are `floorplan`. Adding the following code to the [plans tutorial](https://docs.thatopen.com/Tutorials/Components/Front/Plans):

```ts
const clipper = components.get(OBC.Clipper);
clipper.createFromNormalAndCoplanarPoint(
  world,
  new THREE.Vector3(1, 0, 0),
  new THREE.Vector3(),
);

window.onkeydown = () => {
  clipper.deleteAll(new Set(["default"]));
};
```

I get this:

https://github.com/user-attachments/assets/a2e12e77-9239-4343-aafb-c0e6880100cf

This would also allow you to define your own custom plane types, and delete them all or skip their deletion selectively. I'll publish later today. Let me know if it doesn't work!
